### PR TITLE
Defensive checks for ml labels in config

### PR DESCRIPTION
**Pull request by sushidelivery**
This PR implements a verification process to ensure that each classifier label corresponds with the appropriate model and acit files.

### CODE REVIEW

**Comment by sushidelivery** [comment]
Hello @9and3 ! Just a little note that this PR needs to be checked, no need to merge it yet! Thank you!

**Review by 9and3 — COMMENTED** [review]
Try to check it with some manually added wrong data in the config, e.g.:
- wrong name labels classifier
- wrong name model zenodo
- missing labels classifier

And see if this doesn't break your implementation.

**Comment by sushidelivery** [comment]
Hey @9and3 ! 
- wrong name labels classifier and missing labels classifier - fixed in https://github.com/ibois-epfl/TTool/pull/20/commits/5f270fec330fdc4da6f549803010f8b76ad20eb2

- wrong name model zenodo - fixed in https://github.com/ibois-epfl/TTool/pull/20/commits/6d123a71c3775c9035d7daf2450255f8640395ec

Please, have a look! Thank you !

**Comment by sushidelivery** [comment]
Checked this branch on NUC, seems to be working fine!

File: include/ttool.hh [hunk 1]
```diff
@@ -50,6 +50,7 @@ namespace ttool
         TTool(std::string ttoolRootPath, std::string configFile, std::string cameraCalibFile)
         {
             InitializeConfig(ttoolRootPath, configFile);
+            m_ConfigPtr->CheckAcitFiles(ttoolRootPath);
```
Comment (9and3) [inline include/ttool.hh:53]: As discussed, I think `m_ConfigPtr->CheckAcitFiles(ttoolRootPath);` should not be exposed in the constructor. 
I propose simply to put it in the definition of `InitializeConfig(..)` function, in the same `ttool.hh` header. Here:

https://github.com/ibois-epfl/TTool/blob/6e19a12bcba18326c3bfa8cd76d3e66bd2c27488/include/ttool.hh#L433C9-L438C10

```cpp
 void InitializeConfig(std::string ttoolRootPath, std::string configFile)
        {
            m_ConfigFile = configFile;
            m_ConfigPtr = std::make_shared<ttool::Config>(configFile);
            m_ConfigPtr->SetTToolRootPath(ttoolRootPath);

            # what I propose
            m_ConfigPtr->CheckAcitFiles(ttoolRootPath);
            m_ConfigPtr->CheckClassifierLabelsConfig();

        }
```
Let me know if it works, and give it a go on a compiled AC just to be 100% sure that there are not problems.

**Review by 9and3 — CHANGES_REQUESTED** [review]
We are almost there. I proposed I tiny modification that I think won't alter anything but it's just nicer and tider. Please have a look and let me know when it's done (should not take long).

**Comment by sushidelivery** [comment]
Hey @9and3 ! Done it as you proposed, thank you so much!  Please, have a look! 
P.S. I tested the branch on NUC already.

**Pull request merged**